src/ansi-c/ansi_c_entry_point.cpp

```diff
@@ -304,11 +304,13 @@ bool generate_ansi_c_start_function(
       const symbolt &argv_symbol=ns.lookup("argv'");
 
       {
-        // assume argc is at least one
-        exprt one=from_integer(1, argc_symbol.type);
+        // Assume argc is at least zero. Note that we don't assume it's
+        // at least one, since this isn't guaranteed, as exemplified by
+        // https://www.qualys.com/2022/01/25/cve-2021-4034/pwnkit.txt
```

Comment:
Perhaps add that the C standard only guarantees "The value of argc shall be nonnegative." together with "argv[argc] shall be a null pointer." but nothing beyond that.

Reply:
done